### PR TITLE
(FM-7439) Add RSAPI provider for domain_name

### PR DESCRIPTION
**Pull request by da-ar**

### CODE REVIEW

**Comment by da-ar** [comment]
needs acceptance tests to be updated to the new style
needs: PR #4  to go in

File: spec/unit/puppet/provider/domain_name/nexus_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      expect(Cisco::DomainName).to receive(:new).with('test.puppet.com')
+
+      provider.create(context, 'test.puppet.com', name: 'test.puppet.com', ensure: 'present')
+    end
+  end
+
+  describe 'delete(context, name)' do
+    it 'deletes the resource' do
+      expect(Cisco::DomainName).to receive(:domainnames).and_return(domainnames)
+      expect(domain).to receive(:destroy)
```
Comment (DavidS) [inline spec/unit/puppet/provider/domain_name/nexus_spec.rb]: this is not ascertaining that the correct domain is destroyed

**Comment by DavidS** [comment]
:+1:  except for the hiccup in the unit test.

**Comment by da-ar** [comment]
> 👍 except for the hiccup in the unit test.

yayyy

File: lib/puppet/provider/domain_name/nexus.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+require 'puppet/resource_api/simple_provider'
+require 'cisco_node_utils'
+
+class Puppet::Provider::DomainName::Nexus <  Puppet::ResourceApi::SimpleProvider
```
Comment (DavidS) [inline lib/puppet/provider/domain_name/nexus.rb]: and we just updated the device type to be `cisco_nexus`, so the provider should be called `::CiscoNexus` now. See #8.

File: lib/puppet/provider/domain_name/cisco_nexus.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+require 'puppet/resource_api/simple_provider'
+require 'cisco_node_utils'
+
+class Puppet::Provider::DomainName::CiscoNexus <  Puppet::ResourceApi::SimpleProvider
```
Comment (Thomas-Franklin) [inline lib/puppet/provider/domain_name/cisco_nexus.rb:4]:
```
Error: Could not autoload puppet/type/domain_name: Could not autoload puppet/provider/domain_name/cisco_nexus: uninitialized constant Puppet::Provider::DomainName
Did you mean?  DomainName
```

Which indicates it is expecting `class DomainName::CiscoName` which is weird. Maybe a provider name clash issue?

**Comment by Thomas-Franklin** [comment]
There is a `tests/beaker_tests/netdev_stdlib/domain_name_provider_defaults.rb` and `tests/beaker_tests/netdev_stdlib/domain_namelib.rb` that could probably be removed as the tests have been updated which should work for both modes.

**Review by Thomas-Franklin — APPROVED** [review]

**Pull request merged**